nessie/src/main/java/org/apache/iceberg/nessie/TableReference.java

```diff
@@ -83,12 +95,49 @@ public static TableReference parse(String path) {
     }
 
     if (path.contains("#")) {
-      throw new IllegalArgumentException("Invalid table name:" +
-          " # is not allowed (reference by timestamp is not supported)");
+      String[] tableTimestamp = path.split("#");
+      TableIdentifier identifier = TableIdentifier.parse(tableTimestamp[0]);
+      return new TableReference(identifier, parseTimestamp(tableTimestamp[1]), null);
     }
 
     TableIdentifier identifier = TableIdentifier.parse(path);
 
     return new TableReference(identifier, null, null);
   }
+
+  private enum FormatOptions {
+    DATE_TIME(DateTimeFormatter.ISO_DATE_TIME, Instant::from),
+    LOCAL_DATE_TIME(DateTimeFormatter.ISO_LOCAL_DATE_TIME, t -> LocalDateTime.from(t).atZone(UTC).toInstant()),
```

Comment:
These references are all in the JVM's local time zone?

Reply:
These are to parse the given ISO timestamp into an Instant. If the given timestamp has a timezone on it it will be parsed by `DATE_TIME` with the timezone and relocated to UTC for the instant. If it doesn't have a timezone it will be parsed as a UTC timestamp and if it doesn't have a time at all it will be given midnight at UTC for that date. I believe this is what happens in the tests, I reran w/ a variety of interesting timezones to verify.